src/Microsoft.AspNet.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -150,23 +151,21 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                 await Options.Events.AuthenticationFailed(authenticationFailedContext);
                 if (authenticationFailedContext.HandledResponse)
                 {
-                    return authenticationFailedContext.AuthenticationTicket;
+                    return AuthenticateResult.Success(authenticationFailedContext.AuthenticationTicket);
                 }
-
                 if (authenticationFailedContext.Skipped)
                 {
-                    return null;
+                    return AuthenticateResult.Success(ticket: null);
                 }
 
                 throw;
```

Comment:
Yeah once everyone is more or less happy with the overall design. I will make the final pass through every middleware to address each specific error flow. I updated a few already.